spec/sailor.spec.js

```diff
@@ -176,20 +142,33 @@ describe('Sailor', function () {
                     expect(fakeAMQPConnection.ack.callCount).toEqual(1);
                     expect(fakeAMQPConnection.ack.calls[0].args[0]).toEqual(message);
                 })
-                .done(done, done.fail);
+                .done(done, done); //todo: use done.fail after migration to Jasmine 2.x
         });
 
         it('should send request to API server to update keys', function (done) {
-
+            settings.FUNCTION = 'keys_trigger';
             var sailor = new Sailor(settings);
 
-            spyOn(sailor, "getStepInfo").andReturn({
-                function: "keys_trigger"
-            });
-
-            var nockScope = nock('http://apihost.com:80')
-                .put('/v1/accounts/1234567890', {keys: {oauth: {access_token: 'newAccessToken'}}})
-                .reply(200, "Success");
+            sailor.apiClient = {
```

Comment:
At the end you want to be sure your request work. I you fucked up your request body during a refactoring, you won't know it when mocking the rest client at all. But with a nocking white box testing you have and advantage that nock will tell you about invalid request. However, I might be to paranoid.